### DISCUSSION TITLE
An api method to assign a unique id to a peripheral

### DISCUSSION THREAD

**Discussion by Classified3939** [cc-tweaked/CC-Tweaked#1404] [Q&A]
Within CC, any new computer, turtle, or pocket computer is given a unique id via a class called IdAssigner. However, this class resides within dan200.computercraft.shared.util, not the dan200.computercraft.api package that is typically exposed to mod makers.

If a modded peripheral needs a unique id to store data, and could have multiples placed connected to a single computer (and therefore not able to just use the computer's id), this requires some recreation of code that already exists within computercraft.

Therefore, I would ask that the IdAssigner function getNextId() be exposed to the api in some fashion, probably under the peripheral subheading.

(note: the main page said to ask for functionality in the api package if this exact scenario happens, so this is me doing that)

**Comment by SquidDev** [comment]
You're probably looking for [`ComputerCraftAPI.createUniqueNumberedSaveDir`](https://tweaked.cc/mc-1.19.x/javadoc/dan200/computercraft/api/ComputerCraftAPI.html#createUniqueNumberedSaveDir(net.minecraft.server.MinecraftServer,java.lang.String)) - that should do what you want.

That said, what sort of data are you looking to store here? `createUniqueNumberedSaveDir` makes sense for a file tree, but normally I'd expect peripheral data just to be stored in the block's NBT.
